django_outbox_pattern/consumers.py

```diff
@@ -34,30 +34,39 @@ def __init__(self, connection, username, passcode):
         self.set_listener(self.listener_name, self.listener_class(self))
 
     def message_handler(self, body, headers):
+        db.close_old_connections()
         try:
             body = json.loads(body)
         except json.JSONDecodeError as exc:
             logger.exception(exc)
 
         payload = Payload(self.connection, body, headers)
-        received, created = self.received_class.objects.get_or_create(msg_id=_get_msg_id(headers))
-        if created:
-            received.body = body
-            received.headers = headers
-            try:
-                self.callback(payload)
-            except Exception as exc:  # pylint: disable=broad-except
-                received.status = StatusChoice.FAILED
-                if self.is_connected():
-                    payload.nack()
-                logger.exception(exc)
-            else:
-                received.status = StatusChoice.SUCCEEDED
-                payload.ack()
-            finally:
-                received.save()
-        else:
+        message_id = _get_msg_id(headers)
+
+        if self.received_class.objects.filter(msg_id=message_id).exists():
+            db.close_old_connections()
+            logger.info(f"Message with msg_id: {message_id} already exists. discarding the message")
             payload.ack()
+            return
+
+        received = self.received_class(body=body, headers=headers, msg_id=message_id)
+
+        payload.message = received
+
+        try:
+            self.callback(payload)
+            if payload.saved:
+                payload.ack()
+            elif not payload.saved or not payload.nacked:
+                logger.warning(
+                    f"The save or nack command was not executed, and the routine finished running without receiving an acknowledgement or a negative acknowledgement. message-id: {message_id}"  # noqa: E501 pylint: disable=C0301
+                )
```

Comment:
[suggestion]
We can break the line and pass the parameter as follows (so we don't ignore these errors):

```suggestion
logger.warning(
    "The save or nack command was not executed, and the routine finished running "
    "without receiving an acknowledgement or a negative acknowledgement. "
    "message-id: %s",
    message_id
)
```